scim/usersRecords.bal

```diff
@@ -0,0 +1,454 @@
+    string userType?;
+    string title?;
+    string preferredLanguage?;
+    string locale?;
+    string timezone?;
+    boolean active?;
+    string password?;
+    Group[] groups?;
+    Certificate[] x509Certificates?;
+    Meta meta?;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    Custom urn\:scim\:wso2\:schema?;
+};
+
+# Represents the SCIM User resource request body used for createUser.
+#
+# + schemas - URIs of the used SCIM schemas    
+# + externalId - Identifier by the provisioning client  
+# + userName - Unique identifier within the system  
+# + name - Components of the user's name  
+# + displayName - Name displayed to end-users  
+# + nickName - Casual name of the user 
+# + profileUrl - URI of the user's online profile  
+# + emails - Email addresses for the User 
+# + addresses - Physical mailing address for the user  
+# + phoneNumbers - Phone numbers for the user  
+# + ims - Instant messaging address for the user  
+# + roles - List of roles for the user  
+# + photos - URI of image of the user  
+# + userType - Relationship between the organization and the user  
+# + title - Title of the user
+# + preferredLanguage - Preferred written or spoken languages of the user
+# + locale - Default location of the user
+# + timezone - Time zone of the user
+# + active - Administrative status of the user 
+# + password - Password of the user
+# + groups - Groups to which the user belongs  
+# + x509Certificates - Certificates associated with the user 
+# + meta - Metadata of the user  
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - Attributes belong to business or enterprise extension schema
+# + urn\:scim\:wso2\:schema - Attributes belong to WSO2 extension schema
+public type UserCreate record {
+    string[]|string schemas?;
+    int externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    Email[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Ims[] ims?;
+    Role[] roles?;
+    Photo[] photos?;
+    string userType?;
+    string title?;
+    string preferredLanguage?;
+    string locale?;
+    string timezone?;
+    boolean active?;
+    string password;
+    Group[] groups?;
+    Certificate[] x509Certificates?;
+    Meta meta?;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    Custom urn\:scim\:wso2\:schema?;
+};
+
+# Represents the SCIM Enterprise User extensioon resource.
+#
+# + employeeNumber - Identifier in association with the organization.  
+# + costCenter - Name of a cost center  
+# + organization - Name of an organization  
+# + division - Name of a division  
+# + department - Name of a department  
+# + manager - Manager of the user  
+# + verifyEmail - Email address has been verified or not
+public type SCIMEnterpriseUser record {
+    string employeeNumber?;
+    int costCenter?;
```

Comment:
cost center needs to be a string https://www.rfc-editor.org/rfc/rfc7643.html#:~:text=%22name%22%20%3A%20%22costCenter%22%2C%0A%20%20%20%20%20%20%20%20%22type%22%20%3A%20%22string%22